### PR TITLE
RUN-3305 check uuid in both applications and external-applications

### PR DESCRIPTION
**Pull request by weiteho**
when creating application or external-application, need to check if specified uuid exists in both places.

### CODE REVIEW

File: src/browser/api_protocol/api_handlers/authorization.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ function addPendingAuthentication(uuid, token, file, sponsor, authReqPayload) {
 }
 
 function authenticateUuid(authObj, authRequest, cb) {
-    if (ExternalApplication.getExternalConnectionByUuid(authRequest.uuid)) {
+    if (ExternalApplication.getExternalConnectionByUuid(authRequest.uuid) || coreState.getAppByUuid(authRequest.uuid)) {
```
Comment (rdepena) [inline src/browser/api_protocol/api_handlers/authorization.js]: Hey @weiteho, just want to make sure we take into account the re-connecting scenario, can you confirm that the  core cleans the map on a disconnect ? 

In theory we should be able to connect to a runtime then disconnect and re-connect successfully.

Comment (weiteho) [reply src/browser/api_protocol/api_handlers/authorization.js]: Yes, it's handled already when the connection to server is closed.

Comment (rdepena) [reply src/browser/api_protocol/api_handlers/authorization.js]: 👍  awesome.

**Comment by wenjunche** [comment]
👍

File: src/browser/api_protocol/api_handlers/authorization.js (hunk 1 above)
Comment (whyn07m3) [reply src/browser/api_protocol/api_handlers/authorization.js]: + a test for such scenario?

**Comment by weiteho** [comment]
:white_check_mark: Tests
* [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59c276ba7cb79f732d10c37a)
* [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59c2773e7cb79f732d10c37b)

**Comment by rdepena** [comment]
👍

**Comment by HarsimranSingh** [comment]
Are we waiting on tests to be written?

**Pull request merged**